apps/src/applab/DataProperties.jsx

```diff
@@ -72,20 +68,142 @@ const DataProperties = React.createClass({
   }
 });
 
+const AddKeyRow = Radium(React.createClass({
+  getInitialState() {
+    return {
+      key: '',
+      value: ''
+    };
+  },
+
+  handleKeyChange(event) {
+    this.setState({key: event.target.value});
+  },
+
+  handleValueChange(event) {
+    this.setState({value: event.target.value});
+  },
+
+  handleAdd() {
+    FirebaseStorage.setKeyValue(
+      this.state.key,
+      castValue(this.state.value),
+      () => this.setState(this.getInitialState()),
+      msg => console.warn(msg));
+  },
+
+  render() {
+    return (
+      <tr style={dataStyles.addRow}>
+        <td style={dataStyles.cell}>
+          <input
+              style={dataStyles.input}
+              onChange={this.handleKeyChange}
+              value={this.state.key}
+          ></input>
+        </td>
+        <td style={dataStyles.cell}>
+          <input
+              style={dataStyles.input}
+              onChange={this.handleValueChange}
+              value={this.state.value}
+          ></input>
+        </td>
+        <td style={dataStyles.cell}>
+          <button
+              className="btn btn-primary"
+              style={dataStyles.button}
+              onClick={this.handleAdd}
+          >
+            Add pair
+          </button>
+        </td>
+      </tr>
+    );
+  }
+}));
```

Comment:
As components get more complicated that a simple render function, I would encourage moving them into their own file.